### PR TITLE
Remove sleep; Clean up d2c_messaging_ut nanosleep calls.

### PR DESCRIPTION
**Pull request by JeffMill**

### CODE REVIEW

File: src/utils/d2c_messaging/tests/d2c_messaging_ut.cpp [hunk 1]
```diff
@@ -186,35 +192,15 @@ void* mock_msg_process_thread_routine(void* context)
     // Wait before response.
     if (g_cloudBehavior[g_cloudBehaviorIndex].delayBeforeResponseMS > 999)
     {
-        ADUCPAL_sleep(
-            (static_cast<unsigned int>(g_cloudBehavior[g_cloudBehaviorIndex].delayBeforeResponseMS + 500) / 1000));
+        timespec t;
```
Comment (Nox-MSFT) [inline src/utils/d2c_messaging/tests/d2c_messaging_ut.cpp:195]: For linux os, we can't call nanosleep here if value (ms) > 999.

Per man-page: 

       The value of the nanoseconds field must be in the range 0 to
       999999999.

**Review by Nox-MSFT — CHANGES_REQUESTED** [review]

**Review by yuvalg-MSFT — APPROVED** [review]

File: src/utils/d2c_messaging/tests/d2c_messaging_ut.cpp (hunk 1 above)
Comment (Nox-MSFT) [reply src/utils/d2c_messaging/tests/d2c_messaging_ut.cpp:195]: (When not define ADUCPAL_USE_PAL, ADUCPAL_nanosleep => nanosleep)

Comment (JeffMill) [reply src/utils/d2c_messaging/tests/d2c_messaging_ut.cpp:195]: good catch.  darn it.

Comment (JeffMill) [reply src/utils/d2c_messaging/tests/d2c_messaging_ut.cpp:195]: Weird. I didn't see that here: https://linux.die.net/man/3/nanosleep

Comment (JeffMill) [reply src/utils/d2c_messaging/tests/d2c_messaging_ut.cpp:195]: this works, however:

        ts.tv_sec = 3;
        ts.tv_nsec = 0;
        nanosleep(&ts, NULL);

I'll fix my set_timespec_ms() helper.

**Review by Nox-MSFT — APPROVED** [review]

**Pull request merged**